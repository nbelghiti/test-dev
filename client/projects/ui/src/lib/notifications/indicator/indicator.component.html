<button class="c-notification-indicator">
	<ui-icon name="bell"></ui-icon>

	<span
		*ngIf="count >= 1"
		class="c-notification-indicator__count"
	>
		{{count > 99 ? '99+' : count}}
	</span>
</button>
