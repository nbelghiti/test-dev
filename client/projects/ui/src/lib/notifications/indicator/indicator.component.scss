// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$wrapper-size: 4.4rem;
$indicator-size: 2.4rem;

// =============================================================================
// :: Notification Indicator
// =============================================================================
.c-notification-indicator {
	position: relative;

	display: inline-flex;
	justify-content: center;
	align-items: center;
	width: $wrapper-size;
	height: $wrapper-size;

	font-size: $font-size-base;

	border-radius: 50%;
	border: none;
	background-color: $background-color-grey-96;

	transition:
		background-color $animation-base,
		transform $animation-base;

	cursor: pointer;

	&:hover,
	&:active {
		background-color: $background-color-grey-86;

		transform: scale(0.98) translateY(0.1rem);
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-notification-indicator__count {
	position: absolute;
	top: 0;
	left: 50%;
	z-index: 1;

	width: $indicator-size;
	height: $indicator-size;

	font-weight: $font-weight-ultrabold;

	font-size: $font-size-2x-small;
	line-height: $indicator-size;
	letter-spacing: -0.03rem;
	text-align: center;

	border-radius: 50%;
	background-color: $primary-color;
	color: $white;
}
