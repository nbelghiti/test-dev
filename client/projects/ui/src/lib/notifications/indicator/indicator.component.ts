import { Component, Input } from '@angular/core';

@Component({
	selector: 'ui-notification-indicator',
	templateUrl: './indicator.component.html',
	styleUrls: ['./indicator.component.scss']
})

export class NotificationIndicatorComponent {
	@Input() public count: number;
}
