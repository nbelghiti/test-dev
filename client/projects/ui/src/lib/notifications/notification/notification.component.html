<div
	class="c-notification"
	[ngClass]="{
		'c-notification--success': notificationType.success === type,
		'c-notification--error': notificationType.error === type
	}"
>
	<ui-row contentAfter="true">
		<div class="c-notification__title">
			<div class="c-notification__icon">
				<ui-icon
					[name]="icon"
					size="tiny"
				>
				</ui-icon>
			</div>

			<ui-heading
				[level]="3"
				[title]="title | translate"
			>
			</ui-heading>
		</div>

		<span
			contentAfter
			class="c-notification__time"
			*ngIf="parsedTime"
		>
			{{ parsedTime }}
		</span>
	</ui-row>

	<div class="c-notification__content">
		<ng-content></ng-content>
	</div>

	<button
		*ngIf="!isRead"
		class="c-notification__mark"
		(click)="markRead()"
	>
		Mark as read
	</button>
</div>
