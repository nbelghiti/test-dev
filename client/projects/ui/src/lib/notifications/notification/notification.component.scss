// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$icon-size: 2.4rem;

// =============================================================================
// :: Notification
// =============================================================================
.c-notification {
	@include extend-list-clear();

	padding: 2.5rem 3rem 2rem;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-notification__title {
	display: flex;
	align-items: center;
}

.c-notification__icon {
	display: flex;
	justify-content: center;
	align-items: center;
	width: $icon-size;
	height: $icon-size;
	margin-right: 0.8rem;

	line-height: $icon-size;

	border-radius: 50%;
	background-color: $background-color-grey-96;
}

.c-notification__time {
	font-size: $font-size-1x-small;

	color: $text-color-grey-56;
}

.c-notification__content {
	margin-top: 1rem;
}

.c-notification__mark {
	width: 100%;

	margin-top: 1rem;

	text-align: right;

	color: $primary-color;
}

// =============================================================================
// :: States
// =============================================================================
.c-notification--success {

	.c-notification__icon {
		color: $ui-color-success-dark;
		background-color: $ui-color-success-light;
	}
}

.c-notification--error {

	.c-notification__icon {
		color: $ui-color-error-dark;
		background-color: $ui-color-error-light;
	}
}
