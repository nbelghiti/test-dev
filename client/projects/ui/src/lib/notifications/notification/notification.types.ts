export interface INotificationReadStatus {
	notificationId: string;
	isRead: boolean;
}
