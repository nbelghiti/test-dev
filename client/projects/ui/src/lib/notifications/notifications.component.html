<div
	uiFlyout
	class="c-notifications"
>
	<ui-notification-indicator
		uiFlyoutToggle
		[count]="newAmount"
	>
	</ui-notification-indicator>

	<ul
		uiFlyoutContent
		class="c-notifications__content c-flyout__content--right"
	>
		<ng-content *ngIf="totalAmount > 0; else noNotifications"></ng-content>
	</ul>
</div>

<ng-template #noNotifications>
	<ui-empty-state
		icon="bell-slash"
		[smallIcon]="true"
		message="No notifications"
	>
		No changes have been made to the data yet.
	</ui-empty-state>
</ng-template>
