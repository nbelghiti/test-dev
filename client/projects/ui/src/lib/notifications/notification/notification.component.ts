import { AfterContentInit, Component, EventEmitter, Input, Output } from '@angular/core';

import { IconName } from '../../icon/icon.types';
import { NotificationType } from '../notifications.types';

import { INotificationReadStatus } from './notification.types';

@Component({
	selector: 'ui-notification',
	templateUrl: './notification.component.html',
	styleUrls: ['./notification.component.scss']
})

export class NotificationComponent implements AfterContentInit {
	@Input() public notificationId: string;
	@Input() public isRead: boolean;
	@Input() public title: string;
	@Input() public time: string;
	@Input() public type: NotificationType = NotificationType.success;
	@Input() public icon: IconName = IconName.exclamationCircle;

	@Output() public notificationReadStatus: EventEmitter<INotificationReadStatus> = new EventEmitter<INotificationReadStatus>();

	public notificationType = NotificationType;
	public parsedTime: string;

	public ngAfterContentInit(): void {
		const locale = 'nl-BE';

		if (!this.time) {
			return;
		}

		const timeStamp = new Date(this.time);
		const time = timeStamp.toLocaleTimeString(locale, {
			hour: 'numeric',
			minute: '2-digit'
		});
		const date = timeStamp.toLocaleDateString(locale);

		this.parsedTime = `${this.isToday(timeStamp) ? 'Today' : date.replace(/\//g, '-')} - ${time}`;
	}

	public isToday = (someDate: Date): boolean => {
		const today = new Date();
		return someDate.getDate() === today.getDate() &&
			someDate.getMonth() === today.getMonth() &&
			someDate.getFullYear() === today.getFullYear();
	}

	public markRead(): void {
		this.notificationReadStatus.emit({
			notificationId: this.notificationId,
			isRead: true,
		});
	}
}
