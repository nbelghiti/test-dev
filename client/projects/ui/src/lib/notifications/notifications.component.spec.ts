import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { TranslateModule } from '@ngx-translate/core';

import { EmptyStateComponent } from '../empty-state/empty-state.component';
import { HeadingComponent } from '../heading/heading.component';
import { IconComponent } from '../icon/icon.component';
import { RowComponent } from '../row/row.component';

import { NotificationIndicatorComponent } from './indicator/indicator.component';
import { NotificationComponent } from './notification/notification.component';
import { NotificationsComponent } from './notifications.component';

describe('NotificationsComponent', () => {
	let component: NotificationsComponent;
	let fixture: ComponentFixture<NotificationsComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				TranslateModule.forRoot(),
			],
			declarations: [
				RowComponent,
				HeadingComponent,
				NotificationComponent,
				NotificationsComponent,
				IconComponent,
				EmptyStateComponent,
				NotificationIndicatorComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(NotificationsComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
