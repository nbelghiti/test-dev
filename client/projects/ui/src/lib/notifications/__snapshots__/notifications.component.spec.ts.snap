// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsComponent Should render correctly 1`] = `
<ui-notifications>
  <div
    class="c-notifications"
    uiflyout=""
  >
    <ui-notification-indicator
      uiflyouttoggle=""
    >
      <button
        class="c-notification-indicator"
      >
        <ui-icon
          class="c-icon"
          name="bell"
          ng-reflect-name="bell"
        >
          <i
            class="c-icon__glyph   fas fa-bell"
          />
        </ui-icon>
        
      </button>
    </ui-notification-indicator>
    <ul
      class="c-notifications__content c-flyout__content--right"
      uiflyoutcontent=""
    >
      
      <ui-empty-state
        icon="bell-slash"
        message="No notifications"
        ng-reflect-icon="bell-slash"
        ng-reflect-message="No notifications"
        ng-reflect-small-icon="true"
      >
        <div
          class="c-empty-state"
          ng-reflect-klass="c-empty-state"
          ng-reflect-ng-class="[object Object]"
        >
          <span
            class="c-empty-state__icon c-empty-state__icon--small"
            ng-reflect-klass="c-empty-state__icon"
            ng-reflect-ng-class="[object Object]"
          >
            <i
              class="fas fa-bell-slash"
            />
          </span>
          <p
            class="c-empty-state__message"
          >
            No notifications
          </p>
          <div
            class="c-empty-state__sub-message"
            ng-reflect-ng-class="[object Object]"
          >
             No changes have been made to the data yet. 
          </div>
        </div>
      </ui-empty-state>
    </ul>
  </div>
</ui-notifications>
`;
